### PR TITLE
Add virtual turbine plotting method

### PR DESCRIPTION
**Pull request by paulf81**
**Feature or improvement description**
Some wake models (cc, turbopark) are not yet set up to enable visualization of the horizontal flowfield.  However this visualization can be useful in understanding, sanity checking etc.  This pull request provides a function to calculate the flowfield, through a slower process, of sweeping an extra turbine around the domain in a nested for loop and extracting the u-velocity at hub height. As the turbine is moved throughout the domain, the calculated velocities at the turbine are stored in a local variable and the turbine's wake is not applied to the flow field. For the same resolution, this produces the same results as the `FlorisInterface.calculate_xx_plane` methods. However, this method is significantly slower since it does a full flow field calculation including iterating over each turbine for each visualization grid point.

Pull request comes with demo example to show the new method:
calculate_horizontal_plane_with_turbines

Which at at upper most level behaves much like calculate_horizontal_plane except height can not be freely specified.  Results in plot:
image

**Related issue, if one exists**
Issue #355

### CODE REVIEW

**Comment by Bartdoekemeijer** [comment]
Hi @paulf81. I like this idea of having an artificial turbine in FLORIS function as a probe. I ran the example and saw it run very slowly on my machine. I think this is because of the overhead coming from reinitializing the FLORIS object for every measurement (in the example, we reinitialize FLORIS 600 times). Simply reinitializing floris 600 times on my laptop took `222.75 s`.

Here's an idea based on yours, but circumventing the reinitialize issue. What if we add one turbine for every probe measurement? So if we want to calculate a grid of 40 x 15 wind speeds, we add 600 turbines to the farm. We make these turbines of the `turbine_type=probe`, give them a rotor diameter of `-1`, and a thrust curve of zeros. We can change that probe's location and hub height to specify the (x, y, z) coordinate of every measurement. Then, in `solver.py`, at the very start of the loop: https://github.com/NREL/floris/blob/a23241bb9e45078e36a4662d48c9d3fe0c3316e4/floris/simulation/solver.py#L383

we can add
```
if farm.rotor_diameters[i] < 0.001:
    continue
````

so that these probes have zero effect on the flow field. This would also enable users to handpick measurement locations, rather than always being forced to a grid. It would also resolve the issue with requiring a separate flowfield solver function, like `full_flow_cc_solver`.

@bayc @rafmudaf Feel free to let me know if this is a horrible idea. I understand it would go against some principles in how you designed the FLORIS model.

**Comment by paulf81** [comment]
I think that's really brilliant @Bartdoekemeijer , I really like it!

**Comment by Bartdoekemeijer** [comment]
Okay I just opened a pull request here: https://github.com/paulf81/floris/pull/12

It's a very primitive version of what the turbines-as-probes method that I was trying to explain. It's a little iffy but it mostly works. Try running the example code that I added and see what you think.

**Comment by rafmudaf** [comment]
TODO:
- [ ] Remove example
- [ ] Move plotting function to `floris.tools.visualization`

File: floris/tools/visualization.py [hunk 1]
```diff
@@ -315,3 +319,139 @@ def plot_rotor_values(
...
+                
+                # Save the x and y results
+                x_results[idx] = x
+                y_results[idx] = y
+
+                # Place the test turbine at this location and calculate wake
+                layout_x_test[-1] = x
+                layout_y_test[-1] = y
+                fi.reinitialize(layout=(layout_x_test, layout_y_test))
+                fi.calculate_wake(yaw_angles=yaw_angles)
```
Comment (bayc) [inline floris/tools/visualization.py:433]: I believe we need to down select the yaw angle needed here for the individual turbine.

File: floris/tools/visualization.py [hunk 2]
```diff
@@ -384,3 +388,144 @@ def plot_rotor_values(
...
+        fi = copy.deepcopy(fi_in)
+
+        # If wd/ws not provided, use what is set in fi
+        if wd is None:
+            wd = fi.floris.flow_field.wind_directions
+        if ws is None:
+            ws = fi.floris.flow_field.wind_speeds
+        fi.check_wind_condition_for_viz(wd=wd, ws=ws)
+
+        # # Store the current state for reinitialization
```
Comment (rafmudaf) [inline floris/tools/visualization.py:443]: Is this important to keep?

**Review by rafmudaf — CHANGES_REQUESTED** [review]

File: floris/tools/visualization.py [hunk 3]
```diff
@@ -384,3 +390,144 @@ def plot_rotor_values(
...
+        """
+        Shortcut method to instantiate a :py:class:`~.tools.cut_plane.CutPlane`
+        object containing the velocity field in a horizontal plane cut through
+        using an approximation method of sweeping a turbine around the grid.  Much
+        slower than usual method but helpful for models where the visualization
+        capability is not in place yet
+
+        Args:
+            fi_in (:py:class:`floris.tools.floris_interface.FlorisInterface`):
+                FLORIS object.
```
[the remark this answers is not in the document]
Comment (paulf81) [reply floris/tools/visualization.py:411]: Fixed

File: floris/tools/visualization.py [hunk 4]
```diff
@@ -384,3 +388,144 @@ def plot_rotor_values(
...
+            height (float): Height of cut plane. Defaults to Hub-height.
```
[the remark this answers is not in the document]
Comment (paulf81) [reply floris/tools/visualization.py:412]: Deleted

File: floris/tools/visualization.py [hunk 5]
```diff
@@ -384,3 +388,144 @@ def plot_rotor_values(
...
+                Defaults to 200 points.
+            x_bounds (tuple, optional): Limits of output array (in m).
+                Defaults to None.
+            y_bounds (tuple, optional): Limits of output array (in m).
+                Defaults to None.
+            wd (float, optional): Wind direction for visualization
+                Defaults to None.
+            ws (float, optional): Wind speed for visualization
+                Defaults to None.
+            yaw_angles ((np.ndarray), optional): Wind speed for visualization
```
[the remark this answers is not in the document]
Comment (paulf81) [reply floris/tools/visualization.py:425]: Fixed

**Review by rafmudaf — APPROVED** [review]

**Comment by rafmudaf** [comment]
This PR depends on #569

**Pull request merged**